src/app/(with-sidebar)/myProfile/_hook/useMyProfile.tsx

```diff
@@ -40,23 +63,68 @@ const useMyProfilePage = (): useMyProfilePage => {
   const [skip, setSkip] = useState(0);
   const [hasMorePosts, setHasMorePosts] = useState<boolean | undefined>(true);
   const [post, setPosts] = useState<ArticlesList[]>([]);
+  const [isEditProfileOpen, setIsEditProfileOpen] = useState(false);
+
+  const [userInfo, setUserInfo] = useState<{
+    name: string;
+    bio: string;
+    interests: string[];
+    birthdate: Date;
+    education: string[];
+    achievements: string;
+  }>({
+    name: "",
+    bio: "",
+    interests: [],
+    birthdate: new Date(),
+    education: [],
+    achievements: "",
+  });
 
   const postMutation = trpc.post;
   const likeMutation = trpc.likes;
   const commentMutation = trpc.comments;
+  const userMutation = trpc.user;
+
+  const { data: userDetails } = userMutation.getUserDetails.useQuery(
+    user?.primaryEmailAddress?.emailAddress,
+  );
+
+  const { data: userLikedPosts } = likeMutation.getUserLikedPost.useQuery(
+    {
+      userEmail: user?.primaryEmailAddress?.emailAddress ?? "",
+    },
+    {
+      enabled: !!user?.primaryEmailAddress?.emailAddress,
+    },
+  );
+
+  useEffect(() => {
+    setUserInfo({
+      name: userDetails?.name ?? "",
```

Comment:
why do we need useEffect here? Just pass a prop to MyProfile component and use the prop

Reply:
keeping useEffect for now